tokio/tests/sync_mpsc.rs

```diff
@@ -545,6 +545,56 @@ async fn try_reserve_fails() {
     let _permit = tx.try_reserve().unwrap();
 }
 
+#[maybe_tokio_test]
+async fn try_reserve_many_fails() {
+    for i in 4..20 {
+        let (tx, mut rx) = mpsc::channel(i);
+
+        let mut permit1 = assert_ok!(tx.try_reserve_many(i - 2));
```

Comment:
This test does a lot of stuff, but it's not clear to me that you're covering all the things I mentioned. Instead of putting everything in the same test, I would like to see several really small and easy-to-understand tests for each case that I mentioned.